### PR TITLE
Changelog Update

### PR DESCRIPTION
**Pull request by ColinKYuen**
### Summary

Changelog Update

### Description

Updating CHANGELOG.md to reflect changes made
* Plugin Manager
* Enhanced Failure Monitoring
* Updating Dependencies 
* Replacing JBoss with javassist
* Fixing checkstyle warnings
* Updating `NetworkFailureFailoverIntegrationTest`

### Additional Reviewers

@karenc-bq @sergiyv-bitquill @hsuamz

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
 # Changelog
 
+## [Version 0.4.0 (Public Preview)](https://github.com/awslabs/aws-mysql-jdbc/releases/tag/0.4.0) - 2021-12-14
+
+### Added
+* Connection plugin support through Connection Plugin Manager. More details can be found [here](https://github.com/awslabs/aws-mysql-jdbc#connection-plugin-manager).
+* Connection plugin, [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc/#enhanced-failure-monitoring), to reduce the time required to detect connection failure.
```
Comment (karenc-bq) [inline CHANGELOG.md:7]: I don't think we have to mention connection plugin again.
```suggestion
* [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc/#enhanced-failure-monitoring) to reduce the time required to detect connection failure.
```

Comment (karenc-bq) [reply CHANGELOG.md:7]: It doesn't necessarily "reduce the time required", it kind of depends on user settings.
I prefer something like "[Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc/#enhanced-failure-monitoring) for more sophisticated failure detection."

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+* Tutorial on writing custom connection plugin can be found [here](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/customplugins).
+
+### Improvements
+* NetworkFailuresFailoverIntegrationTest now uses environment variables to match FailoverIntegrationTest and ReplicationFailoverIntegrationTest.
+* Updated all dependencies.
+* Replaced jboss-as-connector with more up-to-date javassist.
+* Fixed checkstyle warnings.
```
Comment (karenc-bq) [inline CHANGELOG.md:14]:
```suggestion
* Addressed checkstyle warnings.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+
+### Added
+* Connection plugin support through Connection Plugin Manager. More details can be found [here](https://github.com/awslabs/aws-mysql-jdbc#connection-plugin-manager).
+* Connection plugin, [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc/#enhanced-failure-monitoring), to reduce the time required to detect connection failure.
+* Tutorial on writing custom connection plugin can be found [here](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/customplugins).
+
+### Improvements
+* NetworkFailuresFailoverIntegrationTest now uses environment variables to match FailoverIntegrationTest and ReplicationFailoverIntegrationTest.
+* Updated all dependencies.
+* Replaced jboss-as-connector with more up-to-date javassist.
```
Comment (karenc-bq) [inline CHANGELOG.md:13]:
```suggestion
* Removed the `jboss-as-connector` dependency.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,20 @@
...
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).
+
+## [Version 0.4.0 (Public Preview)](https://github.com/awslabs/aws-mysql-jdbc/releases/tag/0.4.0) - 2021-12-14
+
+### Added
+* Connection plugin support through Connection Plugin Manager. More details can be found [here](https://github.com/awslabs/aws-mysql-jdbc#connection-plugin-manager).
+* [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc/#enhanced-failure-monitoring) for more sophisticated connection failure detection.
+* Tutorial on writing custom connection plugin can be found [here](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/customplugins).
+
+### Improvements
```
Comment (karenc-bq) [inline CHANGELOG.md:13]: Lets make sure our changelog actually conforms to this format. https://keepachangelog.com/en/1.0.0/
While we are at it lets fix the previous ones too.

**Review by karenc-bq — COMMENTED** [review]
LGTM after adding the changes as discussed offline.

**Review by karenc-bq — APPROVED** [review]

**Review by sergiyv-bitquill — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,30 +1,48 @@
...
+All notable changes to this project will be documented in this file.
 
-## [Version 0.3.0 (Public Preview)](https://github.com/awslabs/aws-mysql-jdbc/releases/tag/0.3.0) - 2021-11-18
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).
+
+## [0.4.0] - 2021-12-14
+
+### Added
+* Connection plugin support through Connection Plugin Manager. More details can be found [here](https://github.com/awslabs/aws-mysql-jdbc#connection-plugin-manager).
+* [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc/#enhanced-failure-monitoring) for more sophisticated connection failure detection.
```
Comment (hsuamz) [inline CHANGELOG.md:10]: [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc/#enhanced-failure-monitoring) for improved failure detection.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,30 +1,48 @@
...
+* Tutorial on writing custom connection plugin can be found [here](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/customplugins).
```
Comment (hsuamz) [inline CHANGELOG.md:11]: I think having the tutorial here is a bit awkward given that it's not a code change in the actual driver. Could we link this to the first point? Tutorial on creating custom connection plugins can be found [here](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/customplugins).

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,30 +1,48 @@
...
+
+### Changed
+* NetworkFailuresFailoverIntegrationTest now uses environment variables to match FailoverIntegrationTest and ReplicationFailoverIntegrationTest.
+* Updated all dependencies.
+
+### Removed
+* `jboss-as-connector` as a dependency.
```
Comment (hsuamz) [inline CHANGELOG.md:18]: Can we provide a short blurb on why we removed jboss?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,30 +1,48 @@
 # Changelog
+All notable changes to this project will be documented in this file.
 
-## [Version 0.3.0 (Public Preview)](https://github.com/awslabs/aws-mysql-jdbc/releases/tag/0.3.0) - 2021-11-18
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).
+
+## [0.4.0] - 2021-12-14
+
+### Added
+* Connection plugin support through Connection Plugin Manager. More details can be found [here](https://github.com/awslabs/aws-mysql-jdbc#connection-plugin-manager).
```
Comment (hsuamz) [inline CHANGELOG.md:9]: Consider "Added the ability to execute additional or supplementary logic related to a `Connection` through the introduction of a Connection Plugin Manager. Additional details can be found [here](https://github.com/awslabs/aws-mysql-jdbc#connection-plugin-manager) and a tutorial on creating custom connection plugins can be found [here](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/customplugins)."

**Review by hsuamz — APPROVED** [review]

**Pull request merged**